### PR TITLE
Feat: Album year + sorting

### LINKED ISSUE #549: [Feature Request] Add album sorting by year
### Solution

I want to be able to sort my albums by year

### Alternatives
Idk

### Additional Context

Well other players most of them do have this feature

### Checklist

-   [x] I have read the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

### PR DESCRIPTION
**Pull request by sevonj**
Closes #549 

## Changes:
- Add year tag to album
- Sort by year in album grid

## Comments:
Null is treated as the lowest value in sorting.

### CODE REVIEW

File: app/src/main/java/io/github/zyrouge/symphony/services/groove/repositories/AlbumRepository.kt [hunk 1]
```diff
@@ -60,6 +61,7 @@ class AlbumRepository(private val symphony: Symphony) {
                     id = albumId,
                     name = song.album!!,
                     artists = song.artists.toMutableSet(),
+                    year = song.year,
```
[the remark this answers is not in the document]
Comment (sevonj) [reply app/src/main/java/io/github/zyrouge/symphony/services/groove/repositories/AlbumRepository.kt:64]: My understanding is that year should always mean the particular release. There's also originalyear.

I've looked into it more, and it became frustratingly hard to find a "correct" way to interpret this data.

[the remark this answers is not in the document]
Comment (sevonj) [reply app/src/main/java/io/github/zyrouge/symphony/services/groove/repositories/AlbumRepository.kt:64]: There are three different years that need to be distinguished:
- Original track release year
- Original album release year
- Album release year (Year of the specific version)

I'm using [Picard](https://github.com/metabrainz/picard) to manage my library. It puts version year to the "YEAR" tag.

Another thought: if we _really_ can't distinguish the dates, how about using a range?

[the remark this answers is not in the document]
Comment (zyrouge) [reply app/src/main/java/io/github/zyrouge/symphony/services/groove/repositories/AlbumRepository.kt:64]: Range sounds good. But for sorting, we should consider the start year. Also, name the properties `startYear` and `endYear`.

**Comment by sevonj** [comment]
Should be OK now:

- I renamed year to startYear, and added endYear.
- They track minimum and maximum values from the songs.

I spent too much time tearing my hair out trying to figure why this doesn't work, but apparently the app just couldn't parse the year from mp3 files I created for testing. Probably the same issue as #182.

There are further improvements to make, such as displaying the year, and tracking original year, but I'd rather break those into separate issues.

**Review by zyrouge — APPROVED** [review]

**Pull request merged**